tardis/plasma/properties/ion_population.py

```diff
@@ -223,31 +224,39 @@ def _calculate_block_ids(phi):
         return calculate_block_ids_from_dataframe(phi)
 
     def calculate(self, phi, partition_function, number_density):
-        n_e_convergence_threshold = 0.05
-        n_electron = number_density.sum(axis=0)
-        n_electron_iterations = 0
-
-        while True:
+        if self._electron_densities is None:
```

Comment:
maybe what you did is fine. I was hoping to make it even nicer - but maybe for now it's fine.